src/services/users.service.ts

```diff
@@ -36,13 +20,20 @@ export class UsersService {
     return this.userRepository.findOne(id);
   }
 
+  async findUserByEmail(email: string) {
+    return await User.createQueryBuilder('user')
+    .addSelect('user.password')
+    .where({ email })
+    .getOneOrFail()
+  }
+
   createUser(user: User) {
-    this.hashUserPassword(user);
-    return this.userRepository.insert(user);
+    //hash user password
+    user.password = hashSync(user.password, genSaltSync());
```

Comment:
Please do not use sync functions, as Node a is a single thread language that is going to block the main thread application if the requests using this functions scale.

The better thing you can do here is to use asynchronous functions in order to execute this in a different queue and only reach the main thread when it's available to start working on new tasks.